server/controllers/finance/patientInvoice.js

```diff
@@ -352,3 +353,16 @@ function safelyDeleteInvoice(guid) {
       return transaction.execute();
     });
 }
+
+function isPosted(req, res, next) {
```

Comment:
Oof.  This is a huge query.  We should probably store the "posted" value in the invoice table - I thought it already existed on there.

Let's do this:
 1. Update the schemas of the `invoice`, `cash` and `voucher` tables to include `posted`.  It should be a NOT NULL 1 or 0 with a default to 0.
 2. Add queries to [the server-side trialbalance controller](https://github.com/IMA-WorldHealth/bhima/blob/master/server/controllers/finance/trialBalance/index.js#L93) to update the `posted` status of the invoices, cash payments, and/or vouchers where their `uuid` is found in the `record_uuid` of the posted records.
 3. We can just use the `posted` value directly from the invoice table then.

That way, we aren't locking the entire general ledger every time we do a lookup of an invoice.